File: src/main/cub/hackerrank/tutorials/a30DaysOfCode/day20/Solution.java
package cub.hackerrank.tutorials.a30DaysOfCode.day20;


import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        for (int a_i = 0; a_i < n; a_i++) {
            a[a_i] = in.nextInt();
        }

        // Write Your Code Here
        int totalNumberOfSwaps = 0;
        for (int i = 0; i < n; i++) {
            int numberOfSwaps = 0;
            for (int j = 0; j < n - 1; j++) {
                if (a[j] > a[j + 1]) {
                    int x = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = x;
                    numberOfSwaps++;
                }
            }
            totalNumberOfSwaps += numberOfSwaps;

            if (numberOfSwaps == 0) {
                break;
            }
        }
        System.out.println("Array is sorted in " + totalNumberOfSwaps + " swaps.");
        System.out.println("First Element: " + a[0]);
        System.out.println("Last Element: " + a[n - 1]);
    }
}

File: src/main/java/cub/hackerrank/tutorials/a30DaysOfCode/day20/Solution.kt
package cub.hackerrank.tutorials.a30DaysOfCode.day20

import java.util.*

fun main(args: Array<String>) {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT. */

    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    val a = IntArray(n)
    for (a_i in 0 until n) {
        a[a_i] = scanner.nextInt()
    }

    var totalNumberOfSwaps = 0
    for (i in 0 until n) {
        var numberOfSwaps = 0
        for (j in 0 until n - 1) {
            if (a[j] > a[j + 1]) {
                a[j] = a[j+1].also { a[j+1] = a[j] }
                numberOfSwaps++
            }
        }
        totalNumberOfSwaps += numberOfSwaps
        if (numberOfSwaps == 0) {
            break
        }
    }

    println("Array is sorted in $totalNumberOfSwaps swaps.")
    println("First Element: " + a[0])
    println("Last Element: " + a[n - 1])
}
